### PR TITLE
Fix `configBasedir` being evaluated case-sensitively on Windows

### PR DESCRIPTION
**Pull request by nolimitdev**
Fixes not working "extends", "plugins" and "customSyntax" in case of small drive letter and backward slashes in configBasedir path. After this fix warn `When linting something other than CSS, you should install an appropriate syntax, e.g. "postcss-html", and use the "customSyntax" option.` is not shown and plugins works.

> Which issue, if any, is this issue related to?

Such as https://github.com/stylelint/stylelint/issues/5594 but for configBasedir

> Is there anything in the PR that needs further explanation?

e.g. "No, it's self-explanatory."

### CODE REVIEW

File: lib/standalone.js [hunk 1]
```diff
@@ -85,6 +85,17 @@ async function standalone({
 		return Promise.reject(error);
 	}
 
+	// convert e.g. c:\foo\bar to C:/foo/bar to avoid not working "extends", "plugins" and "customSyntax"
+	if (configBasedir) {
+		if (configBasedir[0] && configBasedir[1] === ':') {
```
Comment (Mouvedia) [inline lib/standalone.js:90]: opinion

1. organisation: the path handler/fixer should probably be in a helper/file in `lib/utils/`.
2. implementation: we should probably use builtin fs methods—maybe `fs.realpathSync.native`?—or an existing npm package.
3. consistency: all known/similar cases need to be fixed at the same time

File: .changeset/long-toys-boil.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'stylelint': patch
+---
+
+Fixes not working "extends", "plugins" and "customSyntax" in case of small drive letter and backward slashes in configBasedir path.
```
Comment (ybiquitous) [inline .changeset/long-toys-boil.md:5]: [suggest] Please remove the duplicate changelog.

File: lib/standalone.js (hunk 1 above)
Comment (ybiquitous) [reply lib/standalone.js:90]: [ask] Can we use [`normalize-path`](https://www.npmjs.com/package/normalize-path)? We've already used it.

**Review by ybiquitous — CHANGES_REQUESTED** [review]
@nolimitdev Thanks for creating the pull request. Could you add test cases?

File: lib/standalone.js (hunk 1 above)
Comment (nolimitdev) [reply lib/standalone.js]: Yes, `normalize-path` could be used to fix part with backward slashes but when debugging thoroughly again problem is just lower-cased drive letter, so part with changing slashes I have just removed completely

**Comment by nolimitdev** [comment]
@ybiquitous  I have just created demo repo with bug demo: https://github.com/nolimitdev/repo-stylelint-configbasedir-problem

**Comment by ybiquitous** [comment]
@nolimitdev Thanks for providing the demo repo. The root cause is likely that a drive letter becomes lower-cased. I'm unsure which VSCode, VSCode Stylelint extension, or something is responsible.

But we can accept just converting a driver letter. If possible, please check if [`os.platform()`](https://nodejs.org/api/os.html#osplatform) equals `win32`, in order to avoid unexpected side-effects by the conversion.

Also, can you please add a test case for the change to `lib/__tests__/standalone.test.js` like this?

```js
let absoluteConfigBasedir = path.resolve("path/to/config");
absoluteConfigBasedir = absoluteConfigBasedir.charAt(0).toLowerCase() + absoluteConfigBasedir.substring(1);

const { results } = await standalone({
	/* ... */
	configBasedir: absoluteConfigBasedir,
});

// expectations...
```

CI jobs on Windows should pass the case above.

**Comment by nolimitdev** [comment]
Demo repo is based on stylelint v14.16.0 where is a bug. By changelog in v14.16.1 https://github.com/stylelint/stylelint/blob/main/CHANGELOG.md#14161 there was fixing something related by description to this bug `Fixed: customSyntax resolution with configBasedir (https://github.com/stylelint/stylelint/pull/6536) (@ybiquitous])` but when tested demo repo with v14.16.1 `"customSyntax": "postcss-html"` still does not work. But when tested with v15.0.0 it works with any incorrect/nonexistent path for `stylelint.configBasedir` or `stylelint.configBasedir` can be omitted and `"customSyntax": "postcss-html"` works. But in v15 https://github.com/stylelint/stylelint/blob/main/CHANGELOG.md#1500 there is no mention about fixing something about it.

**Comment by ybiquitous** [comment]
@nolimitdev I want to dig into the problem: Stylelint itself or VSCode extension. So, could you please open a new issue with a minimal reproduction repo if you find this PR is not relevant to the second problem?

**Comment by jeddy3** [comment]
Closing as stale.

**Pull request closed without merging**